lib/SILGen/SILGen.cpp

```diff
@@ -1284,22 +1302,22 @@ void SILGenModule::emitDestructor(ClassDecl *cd, DestructorDecl *dd) {
   if (dd->hasBody()) {
     SILDeclRef destroyer(dd, SILDeclRef::Kind::Destroyer);
     SILFunction *f = getFunction(destroyer, ForDefinition);
-    preEmitFunction(destroyer, dd, f, dd);
+    RegularLocation loc(dd);
+    preEmitFunction(destroyer, dd, f, loc);
     PrettyStackTraceSILFunction X("silgen emitDestroyingDestructor", f);
     SILGenFunction(*this, *f, dd).emitDestroyingDestructor(dd);
-    f->setDebugScope(new (M) SILDebugScope(dd, f));
     postEmitFunction(destroyer, f);
   }
 
   // Emit the deallocating destructor.
   {
     SILDeclRef deallocator(dd, SILDeclRef::Kind::Deallocator);
     SILFunction *f = getFunction(deallocator, ForDefinition);
-    preEmitFunction(deallocator, dd, f, dd);
+    RegularLocation loc(dd);
+    preEmitFunction(deallocator, dd, f, loc);
```

Comment:
I think the deallocating entry point can be marked auto-generated. IIRC the destroyer is the one that contains the user's `deinit` code.

Reply:
I think I saw an emit in emitDeallocatingDestructor. I'll check again.

Reply:
Your right.